server/events/vcs/github_client.go

```diff
@@ -509,19 +509,27 @@ func (g *GithubClient) UpdateChecksStatus(ctx context.Context, request types.Upd
 	}
 
 	status, conclusion := g.resolveChecksStatus(request.State)
-	checkRunOutput := github.CheckRunOutput{
-		Title: &request.StatusName,
-	}
-	if request.Output != "" {
-		checkRunOutput.Text = &output
-	}
 
 	// Update checkrun if it exists and if it's not a rerun
 	// request.state is pending only when an operation starts. So, if the checkrun exists and the state is pending, it is a rerun.
 	if checkRun := g.findCheckRun(request.StatusName, checkRuns); checkRun != nil && request.State != models.PendingCommitStatus {
+
+		// Copy the checkrun output if not passed in.
+		var checkRunOutput github.CheckRunOutput
+		if output != "" {
+			checkRunOutput = github.CheckRunOutput{
+				Title: &request.StatusName,
+				Text:  &output,
+			}
+		} else {
+			checkRunOutput = *checkRun.Output
+		}
```

Comment:
Why do we do this?

Reply:
Github uses the fields in the updateReq object to populate the checkrun. So, for example when we want to update the status of a checkrun which already has some output. In this case, if we don't copy over the output, the checkrun's status will be updated but the output section would be empty. 

For eg: if policy_check is failing for a project, the output section mentions which policies are failing specifically. Now, when the policies are approved, we only set the status to green, which means the output section is cleared out which is not our intended behaviour. So, we copy over the checkrun output if it is not passed in to the updateReq.